reviewing/outdated/src/outdated.ts

```diff
@@ -121,11 +124,13 @@ export async function outdated (
           const { name: packageName } = nameVerFromPkgSnapshot(relativeDepPath, pkgSnapshot)
           const name = dp.parse(relativeDepPath).name ?? packageName
 
+          const userWrittenPref = allDeps[alias]
+          const pref = replaceCatalogProtocolIfNecessary(opts.catalogs ?? {}, alias, userWrittenPref)
```

Comment:
I think something like this would be better as I don't see any reason to pass "userWrittenPref" to the function
```suggestion
          const pref = tryResolveCatalog(opts.catalogs ?? {}, alias) ?? allDeps[alias]
```

Reply:
I think we need to pass `userWrittenPref` for a few reasons?

- The `userWrittenPref` might be something like `catalog:foo`. The `tryResolveCatalog` function will need that to know which catalog to look up.
- If the `userWrittenPref`/`allDeps[alias]` isn't using the `catalog:` protocol, we'll want it to be returned as-is, even if there is a catalog entry for that `alias`.